util/symbols.py

```diff
@@ -21,6 +22,7 @@
 
 # Initialize a spimdisasm context, used to store symbols and functions
 spim_context = spimdisasm.common.Context()
+disassembler = SpimdisasmDisassembler()
```

Comment:
I see you are generating an instance of the disassembler here and in split.py, since the purpose of this PR is to abstract out the disassembler, wouldn't be better to only have one instance everywhere and reference that one instead?

Reply:
I think it sounds fine as long as it doesn't mean changing like every splat function lol. hopefully we can store the disassembler in an object somewhere or something?

Reply:
We have been using the `spim_context` from `symbols.py` as a global all over the place, so I guess we could put the disassembler here and reference this instance for now?

Reply:
How is this? https://github.com/ethteck/splat/pull/234/commits/769f31c8fcc3bace852d4151be74baab88be5364

Reply:
I may be wrong, but explicitly referencing `spimdisasm_disassembler.instance` wouldn't defeat the purpose of abstracting the disassembler out?
I think the instance should have some kind of generic name so the specific disassembler shouldn't really matter for the rest of splat

so, something like
```py
# disassembler_instance.py or whatever

if functionToKnowIfWeShouldUseSpimdisasm():
    # typehint as the abstract class
    disassembler: Disassembler = spimdisasm_disassembler.SpimdisasmDisassembler()
elif functionForOtherDisassembler():
    disassembler: Disassembler = otherdisassembler.otherDis()
else:
    raise # some error, idk
# etc
```
Currently this would initialize the spimdisasm disassembler uncoditionally, but this could be tweaked in the future

It would be used like this:
```py
# split.py

def main(config_path, modes, verbose, use_cache=True, skip_version_check=False):
    global config

    disassembler_instance.disassembler.check_version(skip_version_check, VERSION)
    
    # etc
```
This is what I meant by the rest of splat not caring about the specific disassembler.

What do you think about this idea?

Reply:
Ok, how is this? https://github.com/ethteck/splat/pull/234/commits/0977f3b9f810a9ddb6853e278a30f258457dbead

Reply:
I think that is what I had in mind, yeah.
The only remaining complain I have about it is spimdisasm doesn't have gc support, so you should remove that bit. (yes, splat doesn't have a disassembler for gc yet)

Reply:
Ok, gc removed https://github.com/ethteck/splat/pull/234/commits/486c1b38a0a4d8a17a314cfb064dfddde56e6380